website/client/src/store/actions/tasks.js

```diff
@@ -94,6 +101,7 @@ export async function create (store, createdTask) {
     const list = store.state.tasks.data[type];
 
     sanitizeChecklist(t);
+    updateIsDue(store, t, isUserTask);
```

Comment:
@kirsty-tortoise I was thinking, to avoid moving too much logic to the client what we could do, to avoid new properties not being tracked by vue, is to (like it's done elsewhere) change the task object completely instead of just assigning new properties when the server response comes back, so here at line 114 for example:

```
    const tasksArr = store.state.tasks.data[`${taskRes.type}s`];
    const taskDataIndex =tasksArr.findIndex(t => t._id === taskRes._id);
    if (taskDataIndex !== -1) {
        tasksArr[taskDataIndex] = { ...taskData, ...taskRes }; // works like Object.assign but replaces the object
    }
```

how is that looking?

Reply:
Seems very sensible, I'll change that.

Reply:
could you push the commit?

Reply:
```
    const tasksArr = store.state.tasks.data[`${taskRes.type}s`];
    const taskDataIndex =tasksArr.findIndex(t => t._id === taskRes._id);
    if (taskDataIndex !== -1) {
        tasksArr[taskDataIndex] = { ...tasksArr[taskDataIndex], ...taskRes }; // works like Object.assign but replaces the object
    }
```

there was an error in taskData

Reply:
I've pushed it. It still works fine until I remove line 129 which updates isDue on the client side, and then I seem to be getting the same problem as before with Vue as far as I can tell, so it doesn't seem to be helping with having it properly tracked.

Reply:
Oops sorry - I changed it back for testing purposes and I thought I had posted your code in lines 114-118. I'll change it back in a second. I think it will still work, but if I remove line 129 then it stops working again.

Reply:
I hope that's what you meant?

Reply:
@kirsty-tortoise sorry it took a while but turns out the problem exists only when creating a new task because Vue doesn't recognize new properties (it'll get fixed in Vue 3), I pushed a commit directly https://github.com/HabitRPG/habitica/pull/12522/commits/cbb8be8ea46cacf3366e23dedd28b1502249eeae